sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/jms/PremiumServiceBusJMSAutoConfiguration.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.jms;
+
+import com.microsoft.azure.servicebus.jms.ServiceBusJmsConnectionFactory;
+import com.microsoft.azure.servicebus.jms.ServiceBusJmsConnectionFactorySettings;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnResource;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jms.config.DefaultJmsListenerContainerFactory;
+import org.springframework.jms.config.JmsListenerContainerFactory;
+import org.springframework.jms.connection.CachingConnectionFactory;
+
+import javax.jms.ConnectionFactory;
+
+import static com.azure.spring.utils.ApplicationId.AZURE_SPRING_SERVICE_BUS;
+
+/**
+ * Automatic configuration class of ServiceBusJMS for Premium Service Bus
+ */
+@Configuration
+@ConditionalOnClass(ServiceBusJmsConnectionFactory.class)
+@ConditionalOnResource(resources = "classpath:servicebusjms.enable.config")
+@ConditionalOnExpression(value = "'${spring.jms.servicebus.pricing-tier}'.equalsIgnoreCase('premium')")
+@EnableConfigurationProperties(AzureServiceBusJMSProperties.class)
+public class PremiumServiceBusJMSAutoConfiguration {
```

Comment:
A lot of duplicated code between the 2 configuration. Could you please change to use java code (`if... else ...`) instead of use conditional configuration?